handlers.go

```diff
@@ -184,6 +190,23 @@ func (s *server) logout(w http.ResponseWriter, r *http.Request) {
 		http.Redirect(w, r, "/", http.StatusSeeOther)
 		return
 	}
+
+	// Check if the provider has a revocation_endpoint
+	revocationEndpoint, err := revocationEndpoint(s.provider)
+	if err != nil {
+		logger.Warnf("Error getting provider's revocation_endpoint: %v", err)
+	} else {
+		ctx := setTLSContext(r.Context(), s.caBundle)
+		token := session.Values[userSessionOAuthTokens].(oauth2.Token)
+		err := revokeTokens(ctx, revocationEndpoint, &token, s.oauth2Config.ClientID, s.oauth2Config.ClientSecret)
+		if err != nil {
+			logger.Errorf("Error revoking tokens: %v", err)
+			returnStatus(w, http.StatusInternalServerError, "Failed to revoke access/refresh tokens, please try again")
```

Comment:
I think that we should return a 503 error here instead, since this is temporary.

Reply:
That's the hardest one.
The issue with the current code is that the structured information of the request error is lost.
We have to preserve it somehow.
I added a new error type to wrap this extra info and make it available to outer layers.

Reply:
Ok, looks good.